data_model/src/trigger/set.rs

```diff
@@ -304,17 +312,21 @@ impl Set {
     /// This actions will be inspected in the next [`Set::inspect_matched()`] call
     // Passing by value to follow other `handle_` methods interface
     #[allow(clippy::needless_pass_by_value)]
-    pub fn handle_pipeline_event(&self, event: PipelineEvent) {
+    pub fn handle_pipeline_event(&mut self, event: PipelineEvent) {
+        let mut acc = Vec::new();
```

Comment:
Consider using a smallvec. Stack space is relatively cheap and we avoid quite a bit of indirection.

Reply:
To be fair I think this has to be overhauled anyway. Once we have merged new-sumeragi my attention will be turned towards the WorldStateView + ISI/Wasm execution + triggers.

Reply:
@SamHSmith , what exactly do you think we should revise about triggers?